src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -554,6 +554,58 @@ void LateInlineVirtualCallGenerator::do_late_inline() {
   CallGenerator::do_late_inline_helper();
 }
 
+// replace box node to scalar node only in case it is directly referenced by debug info
+static void replace_box_to_scalar(CallNode* call, Node* resproj) {
+  if (resproj != nullptr && call->is_CallStaticJava() &&
+      call->as_CallStaticJava()->is_boxing_method()) {
+    Unique_Node_List safepoints;
+    for (DUIterator_Fast imax, i = resproj->fast_outs(imax); i < imax; i++) {
+      Node* m = resproj->fast_out(i);
+      if (m->is_SafePoint()
+          && (!m->is_Call() || !m->as_Call()->has_non_debug_use(resproj))) {
+        safepoints.push(m);
+      } else {
+        return;
+      }
+    }
+
+#ifndef PRODUCT
+    if (PrintEliminateAllocations && safepoints.size() > 0) {
```

Comment:
Strictly speaking, `PrintEliminateAllocations` is EA-specific flag.
Not sure it has to cover `EliminateAutoBox`/`AggressiveUnboxing`.

Reply:
I think it is just like  https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/macro.cpp#L1158
or we add a new option here?

Reply:
I missed that `PrintEliminateAllocations` is used in `PhaseMacroExpand::eliminate_boxing_node()`. 

Looks good then.